pydantic/main.py

```diff
@@ -402,28 +415,54 @@ def validate_model(model, input_data: dict, raise_exc=True):  # noqa: C901 (igno
     """
     validate data against a model.
     """
+    def _deprecated_values(config) -> bool:
+        """
```

Comment:
No need for a separate method, also move the depeciation logic to the `__new__` method.

Reply:
Not sure where you mean